server/src-lib/Hasura/GraphQL/RemoteServer.hs

```diff
@@ -217,7 +217,9 @@ instance J.FromJSON (FromIntrospection G.UnionTypeDefinition) where
     kind  <- o .: "kind"
     name  <- o .:  "name"
     desc  <- o .:? "description"
-    possibleTypes <- o .: "possibleTypes"
+    -- TODO: maybe instead of using `RemoteSchemaInputValueDefinition` here
+    -- we can generalize it, like in other places?
+    possibleTypes :: [FromIntrospection (G.ObjectTypeDefinition G.InputValueDefinition)] <- o .: "possibleTypes"
```

Comment:
Why do you need the explicit typing here?

Reply:
The explicit typing is needed here because of the `G.ObjectTypeDefinition` type now being a higher kinded type and the `FromJSON` instance is only defined for `FromIntrospection (G.ObjectTypeDefinition G.InputValueDefinition)`. So, omitting the type leads to an ambiguity error.